### ISSUE TITLE
bug: Annotated tags are not sorted correctly for `griffe check`

### ISSUE THREAD

**Issue by auhliarik** [mkdocstrings/griffe#327]
### Description of the bug

In order to determine latest Git tag for check of API breakages, `griffe` uses command `git tag -l --sort=-committerdate`, see [here](https://github.com/mkdocstrings/griffe/blob/main/src/_griffe/git.py#L68). While this works well for lightweight tags, annotated tags (tags with attached message) don't seem to have field `committerdate`. Therefore, they are sorted after all lightweight tags in alphabetical order (see below).

### To Reproduce

```console
$ git clone git@github.com:mkdocstrings/griffe.git
$ cd griffe
$ git tag -l --sort=-committerdate
1.4.1
1.4.0
...
0.1.0
$ git tag 1.4.2 -m "Annotated tag"
$ git tag -l --sort=-committerdate
1.4.1
1.4.0
...
0.1.0
1.4.2
```

### Expected behavior

I would like `griffe` to determine latest tag in a way that works also for annotated tags.

### Environment information

```bash
griffe --debug-info  # | xclip -selection clipboard
```

- __System__: macOS-14.7-arm64-arm-64bit
- __Python__: cpython 3.11.6 
- __Environment variables__:
- __Installed packages__:
  - `griffe` v1.4.1

**Comment by pawamoy** [comment]
Hi @auhliarik, thank you for the report! I'll look into it :slightly_smiling_face:

**Comment by pawamoy** [comment]
> For sorting purposes, fields with numeric values sort in numeric order
       (objectsize, authordate, committerdate, creatordate, taggerdate). All
       other fields are used to sort in their byte-value order.

Pushing a fix!

**Comment by pawamoy** [comment]
Fixed in v1.5.1.
